File: spring-boot-oauth2-auth-server/src/main/java/com/bittokazi/oauth2/auth/server/app/services/role/RoleService.java
package com.bittokazi.oauth2.auth.server.app.services.role;

import com.bittokazi.oauth2.auth.server.app.models.tenant.Role;
import com.bittokazi.oauth2.auth.server.app.services.base.BaseService;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.http.ResponseEntity;

public interface RoleService extends BaseService {
	public Object getRoles(Integer page, Integer count);

	public Object addRole(Role role, HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse);

	public ResponseEntity<?> getRole(String id);

	public Object getAllRoles();

	Object updateRole(Role modifiedRole, String id, HttpServletResponse response);

	Object getRoleByName(Role role);
}

File: spring-boot-oauth2-auth-server/src/main/kotlin/com/bittokazi/oauth2/auth/server/app/services/role/RoleService.kt
package com.bittokazi.oauth2.auth.server.app.services.role

import com.bittokazi.oauth2.auth.server.app.models.tenant.Role
import com.bittokazi.oauth2.auth.server.app.services.base.BaseService
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.http.ResponseEntity

interface RoleService : BaseService {
    fun getRoles(page: Int, count: Int): Any

    fun addRole(role: Role, httpServletRequest: HttpServletRequest, httpServletResponse: HttpServletResponse): Any

    fun getRole(id: String): ResponseEntity<*>

	fun allRoles(): Any

    fun updateRole(modifiedRole: Role, id: String, response: HttpServletResponse): Any

    fun getRoleByName(role: Role): Any
}
